mlflow/pipelines/steps/train.py

```diff
@@ -158,6 +172,103 @@ def _run(self, output_directory):
             raw_validation_df, prediction_result, self.target_col
         )
 
+        from mlflow.entities import ViewType
+
+        mlflow_client = MlflowClient()
+        exp_id = _get_experiment_id()
+
+        primary_metric_greater_is_better = self.metric_greater_is_better[self.primary_metric]
+
+        primary_metric_order = "DESC" if primary_metric_greater_is_better else "ASC"
+
+        search_max_results = 100
+        search_result = mlflow_client.search_runs(
+            experiment_ids=exp_id,
+            run_view_type=ViewType.ALL,
```

Comment:
Just to confirm, we also want to include deleted runs?

Reply:
How to set view type to exclude deleted runs but keep all other runs ?

Reply:
See https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.search_runs

Reply:
Agree with Haru that we should just search `ACTIVE_ONLY` runs.

Reply:
Done!

Reply:
Just to confirm, if a historical run has been terminated, but it haven't been deleted, is it an active run ?

Reply:
Yep! :)

Reply:
I was confused because `mlflow.active_run()` return the current active run which haven't ended. But the search_runs API "active_run" type also includes runs which has finished. @harupy